### PR TITLE
Add country code to date detector

### PR DESCRIPTION
**Pull request by ruthvik-17**
added locale to modify detector preferences wrt the user's country.

### CODE REVIEW

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 1]
```diff
@@ -1375,13 +1394,18 @@ def _date_identification_everyday_except_weekends(self, date_list=None, original
...
         patterns = regex_pattern.findall(self.processed_text.lower())
-
+        is_everyday_result = []
         if not patterns:
             weekday_regex_pattern = re.compile(r'\b((week\s?days?|all\sweekdays))\b')
             patterns = weekday_regex_pattern.findall(self.processed_text.lower())
+            every_weekday_pattern = re.compile(r'\b((every|daily|recur|always|continue|every\s*day|all)/s+'
+                                               r'(week\s?days?|all\sweekdays))\b', re.IGNORECASE)
+            is_everyday_result = every_weekday_pattern.findall(self.processed_text.lower())
+            print('printing is_everyday_result in except weekends', is_everyday_result)
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: E1601 [1;31mprint statement used[0m

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 2]
```diff
@@ -1453,13 +1477,19 @@ def _date_identification_everyday_except_weekdays(self, date_list=None, original
...
         regex_pattern = re.compile(r'\b((every\s?day|daily|all\s?days)\s+except\s+weekdays?)\b')
         patterns = regex_pattern.findall(self.processed_text.lower())
+        is_everyday_result = []
         if not patterns:
             weekend_regex_pattern = re.compile(r'\b((week\s?ends?|all\sweekends))\b')
             patterns = weekend_regex_pattern.findall(self.processed_text.lower())
+            every_weekend_pattern = re.compile(r'\b(every|daily|recur|always|continue|every\s*day|all)'
+                                               r'\s+((week\s?ends?|all\sweekends))\b', re.IGNORECASE)
+            is_everyday_result = every_weekend_pattern.findall(self.processed_text.lower())
+            print('printing is_everyday_result in except weekends', is_everyday_result)
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: E1601 [1;31mprint statement used[0m

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 3]
```diff
@@ -1453,13 +1477,19 @@ def _date_identification_everyday_except_weekdays(self, date_list=None, original
...
         regex_pattern = re.compile(r'\b((every\s?day|daily|all\s?days)\s+except\s+weekdays?)\b')
         patterns = regex_pattern.findall(self.processed_text.lower())
+        is_everyday_result = []
         if not patterns:
             weekend_regex_pattern = re.compile(r'\b((week\s?ends?|all\sweekends))\b')
             patterns = weekend_regex_pattern.findall(self.processed_text.lower())
+            every_weekend_pattern = re.compile(r'\b(every|daily|recur|always|continue|every\s*day|all)'
+                                               r'\s+((week\s?ends?|all\sweekends))\b', re.IGNORECASE)
+            is_everyday_result = every_weekend_pattern.findall(self.text)
+            print('printing is_everyday_result in except weekdays', is_everyday_result)
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: E1601 [1;31mprint statement used[0m

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 4]
```diff
@@ -111,7 +164,8 @@ def detect_date(self, text):
         self.text = " " + text.strip().lower() + " "
         self.processed_text = self.text
         self.tagged_text = self.text
-
+        if self.locale:
+            self.get_country_code_from_locale()
```
Comment (amansrivastava17) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: make this function to return country code

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 5]
```diff
@@ -147,51 +201,18 @@ def get_exact_date(self, date_list, original_list):
...
-        self._update_processed_text(original_list)
-        date_list, original_list = self._tomorrows_date(date_list, original_list)
-        self._update_processed_text(original_list)
-        date_list, original_list = self._yesterdays_date(date_list, original_list)
-        self._update_processed_text(original_list)
-        date_list, original_list = self._day_in_next_week(date_list, original_list)
-        self._update_processed_text(original_list)
-        date_list, original_list = self._day_range_for_nth_week_month(date_list, original_list)
-        self._update_processed_text(original_list)
+        if self.country_code in self.country_date_detector_preferences:
```
Comment (amansrivastava17) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: check if country code is not none

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 6]
```diff
@@ -95,6 +97,57 @@ def __init__(self, entity_name, timezone='UTC', past_date_referenced=False):
         self.month_dictionary = MONTH_DICT
         self.day_dictionary = DAY_DICT
         self.bot_message = None
+        self.locale = locale
+        self.country_code = None
+        self.detector_preferences = [self._gregorian_day_month_year_format,
```
Comment (amansrivastava17) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: rename this to default_detector_preference

File: ner_v2/detectors/temporal/date/en/date_detection.py (hunk 5 above)
Comment (ruthvik-17) [reply ner_v2/detectors/temporal/date/en/date_detection.py]: It shouldn't matter as we are giving all country codes which are not in self.country_date_detector_preferences including 'None' the same detector preferences.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/detectors/temporal/date/en/date_detection.py [hunk 7]
```diff
@@ -285,16 +304,20 @@ def _gregorian_day_month_year_format(self, date_list=None, original_list=None):
...
         for pattern in patterns:
             original = pattern[0]
             dd = int(pattern[1])
             mm = int(pattern[2])
             yy = int(self.normalize_year(pattern[3])) if pattern[3] else self.now_date.year
-            if not pattern[3] and self.timezone.localize(datetime.datetime(year=yy, month=mm, day=dd)) < self.now_date:
-                yy += 1
+            try:
+                # to catch dates which are not possible like "31/11" (october 31st)
+                if not pattern[3] and self.timezone.localize(datetime.datetime(year=yy, month=mm, day=dd)) < self.now_date:
```
Comment (srajdev) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: E501 line too long (123 > 119 characters)

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/detectors/temporal/date/en/date_detection.py (hunk 5 above)
Comment (amansrivastava17) [inline ner_v2/detectors/temporal/date/en/date_detection.py]: check if country_code is not None

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

File: ner_v2/detectors/temporal/date/en/date_detection.py (hunk 5 above)
Comment (ruthvik-17) [reply ner_v2/detectors/temporal/date/en/date_detection.py]: done

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by ruthvik-17** [comment]
retest this please

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by amansrivastava17** [comment]
retest this please

**Comment by ruthvik-17** [comment]
retest this please

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Pull request merged**